### PR TITLE
OCTO-10853-dockerize tests

### PR DESCRIPTION
**Pull request by OlteanuRares**

### CODE REVIEW

**Comment by OlteanuRares** [comment]
this works by running ./run_tests <test python version> (test_py37, test_py38., test_py39) if none provided will default to test_py39
also works with docker compose up <test python version>

**Review by lorandvarga — CHANGES_REQUESTED** [review]

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,61 @@
+version: '3.8'
+
+services:
+  db:
```
Comment (lorandvarga) [inline docker-compose.yml:4]: I don't think pycaption needs a db service.

File: test_requirements.txt [hunk 2]
```diff
@@ -0,0 +1,6 @@
+psycopg2-binary
```
Comment (lorandvarga) [inline test_requirements.txt:1]: probably psycopg2 is not necesarry here. I'd say we use the dependencies from tox.ini

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+      - db
+
+  test_py39:
+    image: python:3.9-slim-bullseye
+    command: sh -c "
+      cd pycaption;
+      pip install --upgrade pip;
+      pip install -r test_requirements.txt;
+      pip install -e .;
+      pytest tests/ -vvvv --junitxml test_report.xml --cov=tests --cov-report xml:coverage.xml;
```
Comment (lorandvarga) [inline docker-compose.yml:52]: Should be taken from tox.ini

File: docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+      "
+    volumes:
+      - .:/pycaption
+    depends_on:
+      - db
+
+volumes:
```
Comment (lorandvarga) [inline docker-compose.yml:59]: We probably don't need any volumes

File: docker-compose.yml [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+  pgdata:
+    driver: local
```
Comment (lorandvarga) [inline docker-compose.yml:61]: We should add python 3.10 as in tox.ini

**Review by lorandvarga — CHANGES_REQUESTED** [review]

File: .github/workflows/main.yml [hunk 6]
```diff
@@ -20,26 +20,21 @@ jobs:
...
-      - name: Install Tox
-        run: pip install tox==3
-      - name: Run Tox
+      - name: Run Test
         id: tests
-        # Run tox using the version of Python in `PATH`
-        run: tox -e py
+        run: |
+          ./run_tests.sh test_${{ matrix.python-version }} 
+          ls -al
```
Comment (lorandvarga) [inline .github/workflows/main.yml:32]: We can remove the ls -al and the |

File: .github/workflows/main.yml [hunk 7]
```diff
@@ -20,26 +20,21 @@ jobs:
...
-        run: tox -e py
+        run: |
+          ./run_tests.sh test_${{ matrix.python-version }} 
+          ls -al 
         continue-on-error: true
 
       - name: Add context info to env
         run: |
+          ls -al 
           sudo apt-get install -y --no-install-recommends libxml-xpath-perl
```
Comment (lorandvarga) [inline .github/workflows/main.yml:38]: we can remove the ls -al

File: setup.py [hunk 8]
```diff
@@ -24,7 +24,7 @@
 
 setup(
     name='pycaption',
-    version='2.1.1',
+    version='2.1.1.dev1',
```
Comment (lorandvarga) [inline setup.py:27]: We can remove the .dev1

File: docker-compose.yml [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+      - .:/pycaption
+
+  test_py311:
+    image: python:3.11-slim-bullseye
+    command: sh -c "
+      cd pycaption;
+      pip install --upgrade pip;
+      pip install -r test_requirements.txt;
+      pip install -e .;
+      pytest -vvvv --color=yes --junit-xml=junit.xml --cov=pycaption --cov-report xml:coverage.xml;
```
Comment (lorandvarga) [inline docker-compose.yml:59]: We should record the results as github artifacts if possible.
See https://docs.github.com/en/actions/using-workflows/storing-workflow-data-as-artifacts

**Review by lorandvarga — CHANGES_REQUESTED** [review]

File: .github/workflows/main.yml [hunk 10]
```diff
@@ -20,26 +20,21 @@ jobs:
     strategy:
```
Comment (lorandvarga) [inline .github/workflows/main.yml:20]: Please rename main.yml into something more meaningful like unit_tests.yml

**Review by lorandvarga — CHANGES_REQUESTED** [review]

**Review by lorandvarga — APPROVED** [review]

**Pull request merged**